app/controllers/conferences_controller.rb

```diff
@@ -2,17 +2,24 @@ class ConferencesController < ApplicationController
   protect_from_forgery with: :null_session
   before_action :respond_to_options
   load_and_authorize_resource find_by: :short_title
-  load_resource :program, through: :conference, singleton: true, except: :index
 
   def index
     @current = Conference.where('end_date >= ?', Date.current).reorder(start_date: :asc)
     @antiquated = @conferences - @current
   end
 
-  def show; end
+  def show
+    # have to change "localhost" to ENV['OSEM_HOSTNAME'] in production
+    check_custom_domain if request.host != 'localhost'
+    @program = @conference.program
+  end
 
   private
 
+  def check_custom_domain
```

Comment:
Maybe a more intentional name for this method would be `load_conference_by_domain` or something like that. We are not checking anything here, really.